src/crdt.ts

```diff
@@ -151,13 +160,104 @@ export class CRDT<T extends DocTypes> {
 
   // if (snap) await this.clock.applyHead(crdtMeta.head, this.clock.head)
 
-  async allDocs(): Promise<{ result: DocUpdate<T>[]; head: ClockHead }> {
-    await this.ready();
-    const result: DocUpdate<T>[] = [];
-    for await (const entry of getAllEntries<T>(this.blockstore, this.clock.head, this.logger)) {
-      result.push(entry);
+  /**
+   * Retrieve the current set of documents.
+   */
+  allDocs<T extends DocTypes>({ waitFor }: { waitFor?: Promise<unknown> } = {}): QueryResponse<T> {
+    const stream = this.#stream.bind(this);
```

Comment:
this is exactly where we should make use of:

#stream = () => { }
declaration.